### PR TITLE
DT 2491: view all roles

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesController.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    )
+  )
+  @ApiOperation(
+    value = "get all Roles.",
+    nickname = "getAllRoles",
+    produces = "application/json"
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(code = 401, message = "Unauthorized.", response = ErrorDetail::class),
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesController.kt:61]: Does this need a 403 too - that mentions the role `ROLE_ROLES_ADMIN`?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesControllerIntTest.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    @Test
+    fun `find page of roles sorting by role code`() {
+      webTestClient.get().uri("/api/roles?page=4&size=3&sort=roleCode")
+        .headers(setAuthorisation("ITAG_USER_ADM", listOf("ROLE_ROLES_ADMIN")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .assertPageOfMany()
+        .jsonPath("$.content[1].roleName")
+        .value<String> { assertThat(it).isEqualTo("Auth Client Management (admin)") }
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesControllerIntTest.kt:68]: Not sure why you need a `.value` here...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/RolesService.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  private val roleRepository: RoleRepository,
+) {
+
+  fun getAllRoles(
+    pageable: Pageable
+  ): Page<Authority> {
+
+    val rolesFilter = RolesFilter(
+      roleCodes = null,
+    )
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/RolesService.kt:23]: Is the implementation of this coming in another PR?

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/RolesService.kt (hunk 3 above)
Comment (SimonMitchellMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/RolesService.kt:23]: later yes, I think we will be adding filter to the roles list page, so I have add it here as an empty filter

**Review by mikehalmamoj — DISMISSED** [review]

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**